rpc/http.go

```diff
@@ -151,6 +152,16 @@ func (srv *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 			http.StatusRequestEntityTooLarge)
 		return
 	}
+
+	ct := r.Header.Get("content-type")
+	mediaType, _, err := mime.ParseMediaType(ct)
+	if err != nil || mediaType != "application/json" {
+		http.Error(w,
+			fmt.Sprintf("invalid content type '%s', only application/json is supported", ct),
```

Comment:
You may want to consider not reflecting untrusted user submitted data here and change the message to "invalid content type, only application/json is supported".

Reply:
Good point, I don't immediately see how an adversary can abuse this but there might be ways.